tx/textual/valuerenderer/message.go

```diff
@@ -79,6 +98,73 @@ func (mr *messageValueRenderer) Format(ctx context.Context, v protoreflect.Value
 	return screens, nil
 }
 
+func (mr *messageValueRenderer) FormatRepeated(ctx context.Context, vr ValueRenderer, v protoreflect.Value, name string) ([]Screen, error) {
+	l := v.List()
+
+	if l == nil {
+		return nil, fmt.Errorf("non-List value")
+	}
+
+	screens := make([]Screen, 1)
+
+	screens[0].Text = fmt.Sprintf("%d %s", l.Len(), formatPluralFieldKind(l.Len(), vr.Kind()))
+
+	for i := 0; i < l.Len(); i++ {
+		subscreens, err := vr.Format(ctx, l.Get(i))
+		if err != nil {
+			return nil, err
+		}
+
+		if len(subscreens) == 0 {
+			return nil, fmt.Errorf("empty rendering")
+		}
+
+		headerScreen := Screen{
+			// <field_name> (<int>/<int>): <value rendered 1st line>
+			Text:   fmt.Sprintf("%s (%d/%d): %s", name, i+1, l.Len(), subscreens[0].Text),
+			Indent: subscreens[0].Indent + 1,
+			Expert: subscreens[0].Expert,
+		}
+		screens = append(screens, headerScreen)
+
+		// <optional value rendered in the next lines>
+		for i := 1; i < len(subscreens); i++ {
+			extraScreen := Screen{
+				Text:   subscreens[i].Text,
+				Indent: subscreens[i].Indent + 1,
+				Expert: subscreens[i].Expert,
+			}
+			screens = append(screens, extraScreen)
+		}
+
+	}
+
+	// End of <field_name>.
+	terminalScreen := Screen{
+		Text: fmt.Sprintf("End of %s", formatPluralFieldKind(l.Len(), name)),
+	}
+	screens = append(screens, terminalScreen)
+	return screens, nil
+}
+
+// formatPluralFieldKind makes an honest attempt at making the kind plural, if
+// the length is not one.  Note: It makes no attempts to handle the various oddities
+// of pluralization.  For instance.. Oddity will become Odditys (instead of Oddities)
+func formatPluralFieldKind(length int, kind string) string {
+	formatted := formatFieldName(kind)
```

Comment:
I know it's not done in your PR, but maybe we can consider renaming this function:

```suggestion
	formatted := toSentenceCase(kind)
```

Each time I read this function, I need to look at its godocs to remember what it does

Reply:
Done in dc079c6... also renamed `formatPluralFieldKind` to be `toPluralKind`